[WIP][quant][graphmode] ConvBn Folding works with bias=None

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #32379 [quant][graphmode] FoldConvBatchNorm2d support shared ClassTypes
* **#32543 [WIP][quant][graphmode] ConvBn Folding works with bias=None**

Summary:
Currently we compile constants directly into the graph, but when we
fold ConvBn, we might need to change a constant bias to some Tensor, and
this is not possible with bias being in the `__constants__` list, so we
need to remove `bias` from the list. This might result in some minor perf
regression but it should be unnoticeable and we can restore the perf after
freeze feature is enabled for production models: https://github.com/pytorch/pytorch/pull/32178

Test Plan:
python test/test_jit.py

Reviewers:
suo, mvz
Subscribers:

Tasks:

Tags:

jerryzh168 committed:
[WIP][quant][graphmode] ConvBn Folding works with bias=None

Summary:
Currently we compile constants directly into the graph, but when we
fold ConvBn, we might need to change a constant bias to some Tensor, and
this is not possible with bias being in the `__constants__` list, so we
need to remove `bias` from the list. This might result in some minor perf
regression but it should be unnoticeable and we can restore the perf after
freeze feature is enabled for production models: https://github.com/pytorch/pytorch/pull/32178

Test Plan:
python test/test_jit.py

Reviewers:
suo, mvz
Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

kostmo commented:
## :pill: CircleCI build failures summary and remediations
As of commit 923ae084:


* **2/2** failures introduced in this PR


## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=923ae08450d1e2b5b36e85bd9626d2740c797de4).

### :detective: 2 new failures recognized by patterns
The following build failures do not appear to be due to upstream breakage:
#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/4325820) pytorch_linux_xenial_py3_6_gcc5_4_ge_config_simple_test (1/2)
**Step:** "Test" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=89934745) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=89934745))
<details>
<summary>
<code>Jan 23 19:19:19 RuntimeError: test_jit_simple failed!</code>
</summary>

```
Jan 23 19:19:18 Ran 2351 tests in 68.791s 
Jan 23 19:19:18  
Jan 23 19:19:18 FAILED (failures=4, errors=36, skipped=79, expected failures=1) 
Jan 23 19:19:18  
Jan 23 19:19:18 Generating XML reports... 
Jan 23 19:19:19 Traceback (most recent call last): 
Jan 23 19:19:19   File "test/run_test.py", line 456, in <module> 
Jan 23 19:19:19     main() 
Jan 23 19:19:19   File "test/run_test.py", line 449, in main 
Jan 23 19:19:19     raise RuntimeError(message) 
Jan 23 19:19:19 RuntimeError: test_jit_simple failed! 
Jan 23 19:19:19 + cleanup 
Jan 23 19:19:19 + retcode=1 
Jan 23 19:19:19 + set +x 
Jan 23 19:19:19 =================== sccache compilation log =================== 
Jan 23 19:19:19 =========== If your build fails, please take a look at the log above for possible reasons =========== 
Jan 23 19:19:19 Compile requests                 2 
Jan 23 19:19:19 Compile requests executed        0 
Jan 23 19:19:19 Cache hits                       0 
Jan 23 19:19:19 Cache misses                     0 
Jan 23 19:19:19 Cache timeouts                   0 
```

</details>

#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/4325817) pytorch_linux_xenial_py3_6_gcc5_4_ge_config_legacy_test (2/2)
**Step:** "Test" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=89935551) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=89935551))
<details>
<summary>
<code>Jan 23 19:20:25 RuntimeError: test_jit_legacy failed!</code>
</summary>

```
Jan 23 19:20:25 Ran 2351 tests in 123.727s 
Jan 23 19:20:25  
Jan 23 19:20:25 FAILED (failures=4, errors=36, skipped=62, expected failures=1) 
Jan 23 19:20:25  
Jan 23 19:20:25 Generating XML reports... 
Jan 23 19:20:25 Traceback (most recent call last): 
Jan 23 19:20:25   File "test/run_test.py", line 456, in <module> 
Jan 23 19:20:25     main() 
Jan 23 19:20:25   File "test/run_test.py", line 449, in main 
Jan 23 19:20:25     raise RuntimeError(message) 
Jan 23 19:20:25 RuntimeError: test_jit_legacy failed! 
Jan 23 19:20:26 + cleanup 
Jan 23 19:20:26 + retcode=1 
Jan 23 19:20:26 + set +x 
Jan 23 19:20:26 =================== sccache compilation log =================== 
Jan 23 19:20:26 =========== If your build fails, please take a look at the log above for possible reasons =========== 
Jan 23 19:20:26 Compile requests               260 
Jan 23 19:20:26 Compile requests executed        0 
Jan 23 19:20:26 Cache hits                       0 
Jan 23 19:20:26 Cache misses                     0 
Jan 23 19:20:26 Cache timeouts                   0 
```

</details>



---

<details><summary>This comment was automatically generated by <a href="https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment">Dr. CI</a> (expand for details).</summary>Follow <a href="https://dr.pytorch.org/admin/comments-opt-out.html">this link to opt-out</a> of these comments for your Pull Requests.

Please report bugs/suggestions on the <a href="https://github.com/kostmo/circleci-failure-tracker/issues">GitHub issue tracker</a>.
</details>

jerryzh168 committed:
Update on "[WIP][quant][graphmode] ConvBn Folding works with bias=None"


Summary:
Currently we compile constants directly into the graph, but when we
fold ConvBn, we might need to change a constant bias to some Tensor, and
this is not possible with bias being in the `__constants__` list, so we
need to remove `bias` from the list. This might result in some minor perf
regression but it should be unnoticeable and we can restore the perf after
freeze feature is enabled for production models: https://github.com/pytorch/pytorch/pull/32178

Test Plan:
python test/test_jit.py

Reviewers:
suo, mvz
Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[WIP][quant][graphmode] ConvBn Folding works with bias=None"


Summary:
Currently we compile constants directly into the graph, but when we
fold ConvBn, we might need to change a constant bias to some Tensor, and
this is not possible with bias being in the `__constants__` list, so we
need to remove `bias` from the list. This might result in some minor perf
regression but it should be unnoticeable and we can restore the perf after
freeze feature is enabled for production models: https://github.com/pytorch/pytorch/pull/32178

Test Plan:
python test/test_jit.py

Reviewers:
suo, mvz
Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

suo reviewed (changes requested):
as discussed offline, we should not do this workaround. It changes the meaning of a very important part of PyTorch in a non-conventional way. I'll look into alternatives that involve fixing the jit

jerryzh168 commented:
deprecated by https://github.com/pytorch/pytorch/pull/32745
